zk_toolbox/zkup/zkup

```diff
@@ -0,0 +1,227 @@
+Options:
+  -p, --path <path>         Specify a local path to install zk_toolbox from. Ignored if --repo is provided.
+  -r, --repo <repo>         GitHub repository to install from (e.g., "matter-labs/zksync-era"). Defaults to "matter-labs/zksync-era".
+  -b, --branch <branch>     Git branch to use when installing from a repository. Ignored if --commit or --version is provided.
+  -c, --commit <commit>     Git commit hash to use when installing from a repository. Ignored if --branch or --version is provided.
+  -v, --version <version>   Git tag to use when installing from a repository. Ignored if --branch or --commit is provided.
+  --skip-zk-supervisor      Skip installation of the zk_supervisor binary.
+  -h, --help                Show this help message and exit.
+
+Examples:
+  $(basename "$0") --repo matter-labs/zksync-era --version 0.1.1
+  $(basename "$0") --path /path/to/local/zk_toolbox --skip-zk-supervisor
+EOF
+}
+
+install_local() {
+  if [ ! -d "$ZKUP_PATH/zk_toolbox" ]; then
+    err "Path $ZKUP_PATH does not contain zk_toolbox"
+    exit 1
+  fi
+
+  if [ -n "$ZKUP_BRANCH" ] || [ -n "$ZKUP_COMMIT" ] || [ -n "$ZKUP_VERSION" ] || [ -n "$ZKUP_REPO" ]; then
+    warn "Ignoring --repo, --branch, --commit and --version arguments when installing from local path"
+  fi
+
+  say "Installing zk_toolbox from $ZKUP_PATH"
+  ensure cd "$ZKUP_PATH"/zk_toolbox
+
+  if [ $ZKUP_SKIP_ZK_SUPERVISOR -eq 1 ]; then
+    BINS=(zk_inception)
+  fi
+
+  for bin in "${BINS[@]}"; do
+    say "Installing $bin"
+    ensure_spinner cargo install --root $ZKT_DIR --path ./crates/$bin --force
+  done
+}
+
+install_from_repo() {
+  if [ -n "$ZKUP_PATH" ]; then
+    warn "Ignoring --path argument when installing from repository"
+  fi
+
+  ZKUP_REPO=${ZKUP_REPO:-"matter-labs/zksync-era"}
+
+  say "Installing zk_toolbox from $ZKUP_REPO"
+
+  if [ $ZKUP_SKIP_ZK_SUPERVISOR -eq 1 ]; then
+    BINS=(zk_inception)
+  fi
+
+  if [ -n "$ZKUP_VERSION" ]; then
+    if [ -n "$ZKUP_COMMIT" ] || [ -n "$ZKUP_BRANCH" ]; then
+      warn "Ignoring --commit and --branch arguments when installing by version"
+    fi
+    ensure cargo install --root $ZKT_DIR --git "https://github.com/$ZKUP_REPO" --tag "zk_toolbox-v$ZKUP_VERSION" --locked "${BINS[@]}" --force
+  elif [ -n "$ZKUP_COMMIT" ]; then
+    if [ -n "$ZKUP_BRANCH" ]; then
+      warn "Ignoring --branch argument when installing by commit"
+    fi
+    ensure cargo install --root $ZKT_DIR --git "https://github.com/$ZKUP_REPO" --rev "$ZKUP_COMMIT" --locked "${BINS[@]}" --force
+  elif [ -n "$ZKUP_BRANCH" ]; then
+    ensure cargo install --root $ZKT_DIR --git "https://github.com/$ZKUP_REPO" --branch "$ZKUP_BRANCH" --locked "${BINS[@]}" --force
+  else
+    ensure cargo install --root $ZKT_DIR --git "https://github.com/$ZKUP_REPO" --locked "${BINS[@]}" --force
+  fi
+}
+
+ensure() {
+  if ! "$@" >/dev/null 2>&1; then
+    err "command failed: $*"
+    exit 1
+  fi
+}
+
+ensure_spinner() {
+  local cmd="$*"
+  "$@" >/dev/null 2>&1 &
```

Comment:
i don't think we have to send everything to dev null during the installation

Reply:
Okay, I will remove the spinner since it wouldn't work